src/server/req_preemptjob.c

```diff
@@ -71,23 +71,29 @@ extern struct server server;
 
 extern time_t time_now;
 
-static int preempt_total = 0;
-static int preempt_index = 0;
-static preempt_job_info *preempt_jobs_list = NULL;
+const static char *preempt_methods[] = {
+	"",
+	"suspend",
+	"checkpoint",
+	"requeue",
+	"delete",
+	""
+    };
 
 /**
  * @brief mark a job preemption as failed
  * @param[in] preempt_preq - the preemption preq from the scheduler
  * @param[in] job_id - the job to mark as failed
  * @return void
  */
-static void job_preempt_fail(struct batch_request *preempt_preq, char *job_id, int code) {
-	snprintf(log_buffer, sizeof(log_buffer), "preemption failed for job (%d)", code);
-	log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_DEBUG, job_id, log_buffer);
+static void job_preempt_fail(struct batch_request *preempt_preq, char *job_id) {
+	int preempt_index = preempt_preq->rq_reply.brp_un.brp_preempt_jobs.count;
+	preempt_job_info *preempt_jobs_list = preempt_jobs_list = preempt_preq->rq_reply.brp_un.brp_preempt_jobs.ppj_list;
+
 	preempt_preq->rq_reply.brp_code = 1;
 	strcpy(preempt_jobs_list[preempt_index].order, "000");
 	sprintf(preempt_jobs_list[preempt_index].job_id, "%s", job_id);
-	preempt_index++;
+	preempt_preq->rq_reply.brp_un.brp_preempt_jobs.count = preempt_index + 1;
```

Comment:
why don't we simply preempt_preq->rq_reply.brp_un.brp_preempt_jobs.count++ instead of using a new variable?